fenix/docs/architecture-overview.md

```diff
@@ -2,168 +2,177 @@
 
-See [mozilla.components.lib.state.Action](https://github.com/mozilla-mobile/android-components/blob/main/components/lib/state/src/main/java/mozilla/components/lib/state/Action.kt)
+#### **Description**
+Maintains a [State](#state), a [Reducer](#reducer) to compute new State, and [Middleware](#middleware). Whenever the Store receives a new [Action](#action) via `store.dispatch(action)`, it will first pass the action through its chain of [Middleware](#middleware). These middleware can initiate side-effects in response, or even consume or change the action. Finally, the Store computes new State using previous State and the new action in the [Reducer](#reducer). The result is then stored as the new State, and published to all consumers of the store.
 
-Created by: [Controller](#controller)
+It is recommended that consumers rely as much as possible on observing State updates from the store instead of reading State directly. This ensures that the most up to date State is always used. This can prevent subtle bugs around call order, as all observers are notified of the same State change before a new change is applied.
 
-Is pushed to: [Store](#store)
+There are several global stores like `AppStore` and `BrowserStore`, as well as Stores scoped to individual screens. Screen-based Stores can be persisted across configuration changes, but are generally created and destroyed during fragment transactions. This means that data that must be shared across Stores should be lifted to a global Store or should be passed as arguments to the new fragment.
 
-#### Description
-Simple data object that carries information about a [State](#state) change to a [Store](#store).  An Action describes _something that happened_, and carries any data relevant to that change. For example, `AccountSettingsFragmentAction.SyncFailed(val time: Long)`, which describes a sync that failed at a specific time.
+Screen-based Stores should be created using [StoreProvider.get](https://github.com/mozilla-mobile/firefox-android/blob/main/fenix/app/src/main/java/org/mozilla/fenix/components/StoreProvider.kt).
 
 -------
 
-### <a name="state"/>State
-#### Overview
-Description of the state of a screen
-
-See [mozilla.components.lib.state.State](https://github.com/mozilla-mobile/android-components/blob/main/components/lib/state/src/main/java/mozilla/components/lib/state/State.kt)
+### <a name="state">State</a>
+#### **Overview**
+Description of the state of a screen or other area of the app.
 
-Referenced by: [Store](#store)
+See [mozilla.components.lib.state.State](https://github.com/mozilla-mobile/firefox-android/blob/main/android-components/components/lib/state/src/main/java/mozilla/components/lib/state/State.kt)
 
-#### Description
+#### **Description**
 Simple, immutable data object that contains all of the backing data required to display a screen. This does not include style details like colors and view sizes, which are handled by the [View](#view).
```

Comment:
💡 More of a general idea, 
to replace this
```This does not include style details like colors and view sizes, which are handled by the [View]```
with
```This should ideally only include kotlin/java data types so this can be easily tested```
which I'm assuming was the reasoning behind not including sizes/colors, I think it's fine to include color data types which are pure kotlin. 
e.g if pocket/wallpapers/some other feature has a backend service that sends the hex color to show before the image has been loaded, I think it's fair to have that color (represented as Color/String/X) in the `UIState`